src/material/schematics/ng-add/setup-project.ts

```diff
@@ -66,41 +68,106 @@ function addAnimationsModule(options: Schema) {
   return async (host: Tree, context: SchematicContext) => {
     const workspace = await getWorkspace(host);
     const project = getProjectFromWorkspace(workspace, options.project);
-    const appModulePath = getAppModulePath(host, getProjectMainFile(project));
-
-    if (options.animations === 'enabled') {
-      // In case the project explicitly uses the NoopAnimationsModule, we should print a warning
-      // message that makes the user aware of the fact that we won't automatically set up
-      // animations. If we would add the BrowserAnimationsModule while the NoopAnimationsModule
-      // is already configured, we would cause unexpected behavior and runtime exceptions.
-      if (hasNgModuleImport(host, appModulePath, noopAnimationsModuleName)) {
-        context.logger.error(
-          `Could not set up "${browserAnimationsModuleName}" ` +
-            `because "${noopAnimationsModuleName}" is already imported.`,
-        );
-        context.logger.info(`Please manually set up browser animations.`);
+
+    try {
+      addAnimationsModuleToNonStandaloneApp(host, project, context, options);
+    } catch (e) {
+      if (e instanceof SchematicsException && e.message.includes('Bootstrap call not found')) {
```

Comment:
I'm worried that "sniffing" on the exception + its error message is rather fragile (will break any time someone decides to alter the error message). Could we have a specific exception type for this case? Or some other reliable way detecting what is going on?

Reply:
I'm not a fan of it either, but I was trying to avoid changing any of the vendored schematic utilities. We have unit tests for this so if the message changes and the check doesn't hold anymore, we'll know about it.

Reply:
Yeah, I guess this comes down to the common utils at some point, but for now seems fine.